File: dynamicpdf-vbnet-examples/Utility/DocumentExampleGenerator.vb
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.PageElements.Html

Namespace DynamicPDFCoreSuite.Utility
    Public Class DocumentExampleGenerator
        Public Shared Sub Generate(ByVal doc As Document)
            For i As Integer = 0 To 4
                Dim pg As New Page(PageSize.Letter)

                Dim width As Single = pg.Dimensions.Width - (pg.Dimensions.LeftMargin * 2)
                Dim height As Single = pg.Dimensions.Height - (pg.Dimensions.TopMargin * 2)

                Dim frmHtmlArea As New HtmlArea(TextGenerator.Generate(), 0, 0, width, height)

                pg.Elements.Add(frmHtmlArea)
                doc.Pages.Add(pg)
            Next
        End Sub
    End Class
End Namespace

File: dynamicpdf-csharp-examples/Utility/DocumentExampleGenerator.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.PageElements.Html;

namespace DynamicPDFCoreSuite.Utility
{
    public class DocumentExampleGenerator
    {
        public static void Generate(Document doc)
        {

            for (int i = 0; i < 5; i++)
            {
                Page pg = new(PageSize.Letter);

                float width = pg.Dimensions.Width - (pg.Dimensions.LeftMargin * 2);
                float height = pg.Dimensions.Height - (pg.Dimensions.TopMargin * 2);

                HtmlArea frmHtmlArea = new(TextGenerator.Generate(), 0, 0, width, height);


                pg.Elements.Add(frmHtmlArea);
                doc.Pages.Add(pg);
            }
        }
    }
}
